File: 1dfroggereasy.py
_, s, m = map(int, input().split())

t = [None]
visited = set()

for x in map(int, input().split()):
    t.append(x)

steps = 0

while True:
    if s < 1:
        result = "left"
        break
    elif s > len(t) - 1:
        result = "right"
        break
    elif s in visited:
        result = "cycle"
        break
    elif t[s] == m:
        result = "magic"
        break

    visited.add(s)
    steps += 1
    s += t[s]

print(result)
print(steps)

File: incomplete/1dfroggereasy.lua
-- Run time error. I have working Python code but I can't
-- figure out what's wrong with the Lua translation
local _, s, m = io.read():match("(%d+)%s+(%d+)%s+(%d+)")
s, m = tonumber(s), tonumber(m)

local t = {}
local visited = {}

for e in string.gmatch(io.read(), "[^%s]+") do
    t[#t + 1] = tonumber(e)
end

local steps = 0

while true do
    if s < 1 then
        result = "left"
        break
    elseif s > #t then
        result = "right"
        break
    elseif visited[s] then
        result = "cycle"
        break
    elseif t[s] == m then
        result = "magic"
        break
    end

    visited[s] = true
    steps = steps + 1
    s = s + t[s]
end

print(result)
print(steps)
